src/clincoded/static/components/collection.js

```diff
@@ -36,370 +49,547 @@ var Collection = module.exports.Collection = createReactClass({
     },
 
-    fetchAll: function (props) {
-        var context = props.context;
-        var communicating;
-        var request = this.state.allRequest;
-        if (request) request.abort();
-        var self = this;
-        if (context.all) {
-            communicating = true;
-            request = this.context.fetch(context.all, {
-                headers: {'Accept': 'application/json'}
-            });
-            request.then(response => {
-                if (!response.ok) throw response;
-                return response.json();
-            })
-            .catch(parseAndLogError.bind(undefined, 'allRequest'))
-            .then(data => {
-                self.extractData({context: data});
-                self.setState({communicating: false});
-            });
-            this.setState({
-                allRequest: request,
-                communicating: true
-            });
-        }
-        return communicating;
-    },
-
-    render: function () {
-        var columns = this.state.columns;
-        var context = this.props.context;
-        var defaultSortOn = this.props.defaultSortOn;
-        var sortOn = this.state.sortOn;
-        var reversed = this.state.reversed;
-        var searchTerm = this.state.searchTerm;
-        this.state.searchTerm = searchTerm;
-        var titles = context.columns || {};
-        var data = this.state.data;
-        var params = url.parse(this.props.href, true).query;
-        var total = context.count || data.rows.length;
-        data.sort(sortOn, reversed);
-        var self = this;
-        var headers = columns.map(function (column, index) {
-            var className;
-            if (index === sortOn) {
-                className = reversed ? "tcell-desc" : "tcell-asc";
-            } else {
-                className = "tcell-sort";
-            }
-            return (
-                <th onClick={self.handleClickHeader} key={index}>
-                    {titles[column] && titles[column]['title'] || column}
-                    <i className={className}></i>
-                </th>
-            );
-        });
-        var searchTermLower = this.state.searchTerm.trim().toLowerCase();
-        var matching = [];
-        var not_matching = [];
-        // Reorder rows so that the nth-child works
-        if (searchTerm) {
-            data.rows.forEach(function (row) {
-                if (row.text.indexOf(searchTermLower) == -1) {
-                    not_matching.push(row);
-                } else {
-                    matching.push(row);
+    /**
+     * Sort rows based on the `sortable` property of `cells` in the rows.
+     * Sets new value of @sortColumn and @reversed to the state
+     * @param {array} rows Data to sort
+     * @param {number} sortColumn Column number to sort
+     * @param {boolean} reverse Used to set in ascending or descending order
+     */
+    sortRows: function (rows, sortColumn, reverse) {
+        const reversed = !!reverse;
```

Comment:
Extra `!` or if intentional, wouldn't it need to be `const reversed = reverse`, let me know!

Reply:
That is what it originally was and I wasn't sure why the previous person did that, so I intentionally just left it. I can probably just change it though.